### ISSUE TITLE
Add way to set `documentURI` on `Document`

### ISSUE THREAD

**Issue by henningko** [b-fuze/deno-dom#111]
Referencing #68, there currently is no way to set the `documentURI` on `Document` ([MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/documentURI)), which is required by modules such as `@mozilla/readability`.

JSDOM does this via an `url` parameter, passed into the constructor with the options:
```js
const dom = new JSDOM(``, {
  url: "https://example.org/",
  referrer: "https://example.com/",
  contentType: "text/html",
  includeNodeLocations: true,
  storageQuota: 10000000
});
```
> url sets the value returned by window.location, document.URL, and document.documentURI, and affects things like resolution of relative URLs within the document and the same-origin restrictions and referrer used while fetching subresources. It defaults to "about:blank".
[see JSDOM](https://github.com/jsdom/jsdom)

**Comment by sno2** [comment]
Deno allows setting the `window.location` via a command-line argument. Would it be best to have these properties dynamically loaded if you access `Node.baseURI`, `Document.documentURI`, `Document.baseURI`, and `Document.URL` which would make Deno ask you to provide a `--location` parameter?  Although, this wouldn't really allow for fine-grained control at runtime so it might still be necessary to have a runtime API for this for those _few_ number of libraries and users that need it.

**Comment by b-fuze** [comment]
I'm thinking of doing something like
```ts
import { denoDom, DOMParser } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
const doc = new DOMParser().parseFromString("<h1>foo</h1>", "text/html");
denoDom.setDocumentMetadata(doc, {
  uri:  "https://example.org/example.html",
  referrer:  "https://example.com/",
});

console.log(doc.documentURI); // prints "https://example.org/example.html"
console.log(doc.referrer); // prints "https://example.com/"
```

**Comment by henningko** [comment]
Yes, that looks good. Wonder if you might want to include options parameter in `parseFromString`:

```js
const doc = new DOMParser().parseFromString("<h1>foo</h1>", {
    encoding: "text/html",
    uri: "https://example.org/example.html",
    referrer: "https://example.com/",
});
```

**Comment by b-fuze** [comment]
Yeah, I'd rather avoid extending the actual DOM API. What I think we can do to make it nicer is return the document from `setDocumentMetadata`. Something like:

```ts
import { denoDom, DOMParser } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";

const doc = denoDom.setDocumentMetadata(
  new DOMParser().parseFromString("<h1>foo</h1>", "text/html"),
  {
    uri: "https://example.org/example.html",
    referrer: "https://example.com/",
  },
);

console.log(doc.documentURI); // prints "https://example.org/example.html"
console.log(doc.referrer); // prints "https://example.com/"
```

**Comment by b-fuze** [comment]
The [document implementation is here](https://github.com/b-fuze/deno-dom/blob/master/src/dom/document.ts#L117) and you can another file for that `denoDom` export probably... I'd like to make that a separate export though, so maybe something like `import { setDocumentMetadata } from "jsr:@b-fuze/deno-dom/util`
